### PR TITLE
fix(get pipelines): don't ask for an API token

### LINKED ISSUE #2327: `jx get pipelines` asking for API token when it should not
### Summary 

With Prow & Knative builds enabled, `jx get pipelines` asks for a Jenkins API token, however there is no Jenkins to ask to.

### Jx version

The output of `jx version` is:

```
NAME               VERSION
jx                 1.3.583
jenkins x platform 0.0.2902
Kubernetes cluster v1.9.7-gke.11
kubectl            v1.12.1
helm client        v2.11.0+g2e55dbe
helm server        v2.11.0+g2e55dbe
git                git version 2.7.4
```

### Kubernetes cluster

What kind of Kubernetes cluster are you using & how did you create it?

### PR DESCRIPTION
**Pull request by amuniz**
When Prow and KNative builds are enabled there is no Jenkins to ask for pipelines. `jx get pipelines` returns now the list of release jobs, to be consistent with `jx start pipeline`.

#### Submitter checklist

- [x] Change is code complete and matches issue description.
- [x] Change is covered by existing or new tests.

#### Description

This fix avoid the API Token requirement on `jx get pipelines` when using knative builds, and also makes the output of `jx get pipelines` consistent with `jx start pipeline`.

```
Name                                     URL LAST_BUILD STATUS DURATION
jenkins-x/dummy/master                   N/A N/A        N/A    N/A
```

#### Which issue this PR fixes

fixes #2327 (PLAT-259)

### CODE REVIEW

**Comment by amuniz** [comment]
Test failure is:

```
--- FAIL: TestStepBuildPackApply (1.94s)
    require.go:794: 
        	Error Trace:	step_buildpack_apply_integration_test.go:74
        	Error:      	Received unexpected error:
        	            	template: myJenkinsfile:42:17: executing "myJenkinsfile" at <.Pipelines.Post.Groo...>: can't evaluate field Post in type jenkinsfile.Pipelines
        	            	failed to write file /tmp/test-step-buildpack-apply123564351/Jenkinsfile
        	            	github.com/jenkins-x/jx/pkg/jenkinsfile.(*CreateJenkinsfileArguments).GenerateJenkinsfile
        	            		/home/jenkins/go/src/github.com/jenkins-x/jx/pkg/jenkinsfile/pipeline.go:473
        	            	github.com/jenkins-x/jx/pkg/jx/cmd.(*CommonOptions).invokeDraftPack
        	            		/home/jenkins/go/src/github.com/jenkins-x/jx/pkg/jx/cmd/common_buildpacks.go:241
        	            	github.com/jenkins-x/jx/pkg/jx/cmd.(*StepBuildPackApplyOptions).Run
        	            		/home/jenkins/go/src/github.com/jenkins-x/jx/pkg/jx/cmd/step_buildpack_apply.go:103
        	            	github.com/jenkins-x/jx/pkg/jx/cmd_test.TestStepBuildPackApply
        	            		/home/jenkins/go/src/github.com/jenkins-x/jx/pkg/jx/cmd/step_buildpack_apply_integration_test.go:73
        	            	testing.tRunner
        	            		/usr/local/go/src/testing/testing.go:827
        	            	runtime.goexit
        	            		/usr/local/go/src/runtime/asm_amd64.s:1333
        	Test:       	TestStepBuildPackApply
        	Messages:   	failed to run step
```

I don't think it is related to this PR, @jstrachan?

File: pkg/jx/cmd/get_pipeline.go [hunk 1]
```diff
@@ -65,36 +70,91 @@ func NewCmdGetPipeline(f Factory, in terminal.FileReader, out terminal.FileWrite
 
 // Run implements this command
 func (o *GetPipelineOptions) Run() error {
-	jenkins, err := o.JenkinsClient()
+
+	_, _, err := o.JXClient()
```
Comment (ccojocar) [inline pkg/jx/cmd/get_pipeline.go]: What is the reason for creating an empty JxClient?

File: pkg/jx/cmd/get_pipeline.go [hunk 2]
```diff
@@ -65,36 +70,91 @@ func NewCmdGetPipeline(f Factory, in terminal.FileReader, out terminal.FileWrite
...
 	if err != nil {
 		return err
 	}
-	jobs, err := jenkins.GetJobs()
+
+	_, _, err = o.KubeClient()
```
Comment (ccojocar) [inline pkg/jx/cmd/get_pipeline.go]: also an empty kuberntes client

Comment (amuniz) [reply pkg/jx/cmd/get_pipeline.go]: It is a way to check jx and k8s connections are ok. I followed the [same approach than other commands](https://github.com/jenkins-x/jx/blob/7a411fca6febc7fdcf6df234a6978faa7e8b5a45/pkg/jx/cmd/start_pipeline.go#L97-L104).

File: pkg/jx/cmd/get_pipeline.go (hunk 1 above)
Comment (amuniz) [reply pkg/jx/cmd/get_pipeline.go]: Answered below.

**Pull request merged**